jgrapht-core/src/main/java/org/jgrapht/alg/chordal/BucketList.java

```diff
@@ -0,0 +1,281 @@
+/*
+ * (C) Copyright 2018, by Timofey Chudakov and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.chordal;
+
+import java.util.*;
+
+/**
+ * Data structure for performing lexicographical breadth-first search.
+ * Allows to add and retrieve vertices from buckets, update their labels
+ *
+ * @param <V> the graph vertex type
+ * @author Timofey Chudakov
+ * @see ChordalGraphInspector
+ * @since 1.8
+ */
+public class BucketList<V> {
```

Comment:
This class is currently just an internal data structure for the ChordalGraphInspector, so in its current form it shouldn't be public.

If you intend it to be reusable as a general LexicographicBFS, then it should be renamed and enhanced accordingly before being made public.